### PR TITLE
[DROOLS-6482] NullSafeDereferencing without predicate misses null che…

### PR DESCRIPTION
**Pull request by tkobayas**
…ck in exec-model

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-6482

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Jenkins run fdb

**Comment by tkobayas** [comment]
Jenkins run fdb

File: drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/drlxparse/SingleDrlxParseSuccess.java [hunk 1]
```diff
@@ -72,6 +74,7 @@
     private boolean combined;
 
     private Optional<Expression> implicitCastExpression = Optional.empty();
+    private List<Expression> nullSafeExpressions = new ArrayList<>();
```
Comment (tkobayas) [inline drools-model/drools-model-compiler/src/main/java/org/drools/modelcompiler/builder/generator/drlxparse/SingleDrlxParseSuccess.java:77]: I take a similar approach to `implicitCastExpression`.

**Comment by tkobayas** [comment]
Jenkins run fdb

**Comment by tkobayas** [comment]
(ubuntu-latest, 11)/(windows-latest, 11) failure isn't related to this PR (found in other PR build. but cannot reproduce on my local end).
```
2021-07-30T01:29:14.8600085Z [ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.103 s <<< FAILURE! - in org.drools.modelcompiler.DowncastTest^M
2021-07-30T01:29:14.8613245Z [ERROR] testDowncast[PATTERN_DSL](org.drools.modelcompiler.DowncastTest)  Time elapsed: 0.071 s  <<< FAILURE!^M
2021-07-30T01:29:14.8614659Z java.lang.AssertionError: ^M
2021-07-30T01:29:14.8616431Z [Message [id=1, level=ERROR, path=src/main/java/defaultpkg/PC7/LambdaPredicateC7132C1CB78C0B6BB21618E128BE427C.java, line=24, column=1141^M
2021-07-30T01:29:14.8662441Z    text=The method abs(int) in the type Math is not applicable for the arguments (BigDecimal)], Message [id=2, level=ERROR, path=src/main/java/defaultpkg/PC7/LambdaPredicateC7132C1CB78C0B6BB21618E128BE427C.java, line=0, column=0^M
2021-07-30T01:29:14.8664421Z    text=Java source of src/main/java/defaultpkg/PC7/LambdaPredicateC7132C1CB78C0B6BB21618E128BE427C.java in error:^M
2021-07-30T01:29:14.8665752Z package defaultpkg.PC7;^M
2021-07-30T01:29:14.8666205Z ^M
2021-07-30T01:29:14.8765364Z import static defaultpkg.RulesEACB0BBC98102A3BB31497B07EDB3CFC.*;^M
2021-07-30T01:29:14.8766927Z import org.drools.modelcompiler.DowncastTest.Product;^M
2021-07-30T01:29:14.8768360Z import org.drools.modelcompiler.DowncastTest.LoanProduct;^M
2021-07-30T01:29:14.8769904Z import org.drools.modelcompiler.DowncastTest.CashProduct;^M
2021-07-30T01:29:14.8771257Z import org.drools.modelcompiler.DowncastTest.Proposal;^M
2021-07-30T01:29:14.8772180Z import defaultpkg.*;^M
2021-07-30T01:29:14.8778679Z import org.drools.modelcompiler.dsl.pattern.D;^M
2021-07-30T01:29:14.8780549Z ^M
2021-07-30T01:29:14.8791282Z @org.drools.compiler.kie.builder.MaterializedLambda()^M
2021-07-30T01:29:14.8801230Z public enum LambdaPredicateC7132C1CB78C0B6BB21618E128BE427C implements org.drools.model.functions.Predicate2<org.drools.modelcompiler.DowncastTest.Proposal, org.drools.modelcompiler.DowncastTest.Proposal>, org.drools.model.functions.HashedExpression {^M
2021-07-30T01:29:14.8804728Z ^M
2021-07-30T01:29:14.8817651Z     INSTANCE;^M
2021-07-30T01:29:14.8818108Z ^M
2021-07-30T01:29:14.8818821Z     public static final String EXPRESSION_HASH = "7BA6F742C2F27BFB31ACFB94B400757C";^M
2021-07-30T01:29:14.8819436Z ^M
2021-07-30T01:29:14.8820046Z     public java.lang.String getExpressionHash() {^M
2021-07-30T01:29:14.8820902Z         return EXPRESSION_HASH;^M
2021-07-30T01:29:14.8821400Z     }^M
2021-07-30T01:29:14.8821750Z ^M
2021-07-30T01:29:14.8822227Z     @Override()^M
2021-07-30T01:29:14.8841903Z     public boolean test(org.drools.modelcompiler.DowncastTest.Proposal _this, org.drools.modelcompiler.DowncastTest.Proposal $cp) throws java.lang.Exception {^M
2021-07-30T01:29:14.8852295Z         return org.drools.modelcompiler.util.EvaluationUtil.greaterThanNumbers(Math.abs(((LoanProduct) _this.getProduct()).getLoanValue().subtract(((CashProduct) $cp.getProduct()).getCashValue()).divide(((CashProduct) $cp.getProduct()).getCashValue())), 3l);^M
2021-07-30T01:29:14.8861284Z     }^M
2021-07-30T01:29:14.8861698Z ^M
2021-07-30T01:29:14.8862153Z     @Override()^M
2021-07-30T01:29:14.8863179Z     public org.drools.model.functions.PredicateInformation predicateInformation() {^M
2021-07-30T01:29:14.8880864Z         return new org.drools.model.functions.PredicateInformation("Math.abs((((LoanProduct)product).loanValue.subtract(((CashProduct)$cp.product).cashValue)).divide(((CashProduct)$cp.product).cashValue)) > 3", "R", "r0.drl");^M
2021-07-30T01:29:14.8882301Z     }^M
2021-07-30T01:29:14.8882736Z }^M
2021-07-30T01:29:14.8883132Z ]]^M
2021-07-30T01:29:14.8883705Z    at org.junit.Assert.fail(Assert.java:89)^M
```

Filed https://issues.redhat.com/browse/DROOLS-6534 for investigation

**Review by lucamolteni — APPROVED** [review]

**Comment by tkobayas** [comment]
All FDB errors are not related to this PR.
```
org.kie.server.integrationtests.jbpm.BARuntimeDataServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ByPassUserTaskServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ContainerFailureRecoveryIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ContainerUpdateIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.CustomDataProcessServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.DocumentServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ExceptionHandlingIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.JobServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.KieServerContainerListFilteringIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ProcessDefinitionIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ProcessSearchServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ProcessServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.ProcessUserTaskServicesSecurityIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.QueryDataServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.QueryInKjarIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.RollingUpdateProcessServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.RollingUpdateUserTaskServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.RuntimeDataServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.SLAComplianceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.StartProcessServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.TaskSearchServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.TimerIntegrationTest.org.kie.server.integrationtests.jbpm.TimerIntegrationTest
org.kie.server.integrationtests.jbpm.UserTaskEscalationIntegrationTest.org.kie.server.integrationtests.jbpm.UserTaskEscalationIntegrationTest
org.kie.server.integrationtests.jbpm.UserTaskServiceAdditionalIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.UserTaskServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.WebServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.WebServiceWrappedParamsIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.admin.ProcessInstanceAdminServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.admin.ProcessInstanceMigrationIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.admin.UserTaskAdminServiceIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.jms.JmsResponseHandlerIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.jms.JmsTransactionsIntegrationTest.initializationError
org.kie.server.integrationtests.jbpm.rest.IdentifierJbpmRestIntegrationTest.org.kie.server.integrationtests.jbpm.rest.IdentifierJbpmRestIntegrationTest
org.kie.server.integrationtests.jbpm.rest.JbpmRestIntegrationTest.org.kie.server.integrationtests.jbpm.rest.JbpmRestIntegrationTest
org.kie.server.integrationtests.jbpm.rest.ProcessServiceRestOnlyIntegrationTest.org.kie.server.integrationtests.jbpm.rest.ProcessServiceRestOnlyIntegrationTest

e.g.
[2021-07-30T06:44:22.941Z] [INFO] Running org.kie.server.integrationtests.jbpm.BARuntimeDataServiceIntegrationTest
[2021-07-30T06:44:22.941Z] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.001 s <<< FAILURE! - in org.kie.server.integrationtests.jbpm.BARuntimeDataServiceIntegrationTest
[2021-07-30T06:44:22.941Z] [ERROR] initializationError(org.kie.server.integrationtests.jbpm.BARuntimeDataServiceIntegrationTest)  Time elapsed: 0 s  <<< ERROR!
[2021-07-30T06:44:22.941Z] java.lang.RuntimeException: Failed to create JMS client configuration!
[2021-07-30T06:44:22.941Z] Caused by: javax.naming.CommunicationException: WFNAM00018: Failed to connect to remote host
[2021-07-30T06:44:22.941Z] Caused by: java.net.ConnectException: Connection refused
```

**Review by pibizza — APPROVED** [review]

**Pull request merged**